### PR TITLE
Bump plutus-core to 1.5.0.0

### PR DESCRIPTION
**Pull request by zliu41**

### CODE REVIEW

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs [hunk 1]
```diff
@@ -433,12 +434,12 @@ transExUnits (ExUnits mem steps) =
 exBudgetToExUnits :: PV1.ExBudget -> Maybe ExUnits
 exBudgetToExUnits (PV1.ExBudget (PV1.ExCPU steps) (PV1.ExMemory memory)) =
   ExUnits
-    <$> safeFromInteger memory
-    <*> safeFromInteger steps
+    <$> safeFromSatInt memory
+    <*> safeFromSatInt steps
   where
-    safeFromInteger :: Integral a => a -> Maybe Natural
```
Comment (zliu41) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs:439]: SatInt no longer has Integral instance

**Comment by zliu41** [comment]
To be done in https://github.com/input-output-hk/cardano-ledger/pull/3365 instead.

**Comment by michaelpj** [comment]
I think we should do this independently anyway. The version bump should be independent from actually starting to use the new features!

File: cabal.project [hunk 2]
```diff
@@ -19,7 +19,7 @@ index-state: 2023-02-22T00:00:00Z
 index-state:
   , hackage.haskell.org 2023-02-22T00:00:00Z
 -- Bump this if you need newer packages from CHaP
-  , cardano-haskell-packages 2023-03-22T19:09:40Z
+  , cardano-haskell-packages 2023-04-17T14:00:00Z
```
Comment (lehins) [inline cabal.project:22]: You need to run `niv update cardano-haskell-packages` to update sources too.

See [CONTRIBUTING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/CONTRIBUTING.md#updating-dependencies)

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts/Data.hs [hunk 3]
```diff
@@ -80,12 +80,6 @@ import GHC.Generics (Generic)
 import NoThunks.Class (NoThunks)
 import qualified PlutusLedgerApi.V1 as PV1 -- NOTE PV1.Data === PV2.Data
 
--- =====================================================================
--- PV1.Data is the type that Plutus expects as data. For both V1 and V2.
--- It is imported from the Plutus package, but it needs a few additional
--- instances to also work in the ledger.
-deriving instance NoThunks PV1.Data
```
Comment (lehins) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts/Data.hs:87]: Removal of an orphan instance requires a major version pump in a package, so it should be set to: `cardano-ledger-alonzo-1.2.0.0`

**Review by lehins — CHANGES_REQUESTED** [review]
Couple of required steps are missing, otherwise looks good.

**Review by lehins — APPROVED** [review]
:+1: Thank you

**Pull request merged**